source/dem/particle_heat_transfer.cc

```diff
@@ -0,0 +1,351 @@
+  const double gas_parameter_m,
+  const double thermal_conductivity_gas,
+  const double maximum_pressure,
+  const double effective_microhardness)
+{
+  const double x_1 = 2.0 * maximum_pressure / effective_microhardness;
+  const double x_2 = 0.03 * maximum_pressure / effective_microhardness;
+
+  // If the values are out of the valid range ]0, 2[ of the erfc^-1 function,
+  // this resistance has no physical meaning. Particles are too close or not
+  // close enough for a microgap between them to make sense. This resistance is
+  // used in parallel with another one, so putting it to infinity ignores it in
+  // the total resistance.
+  if (x_1 >= 2.0 || x_1 <= 0.0)
+    {
+      return INFINITY;
+    }
+
+  const double a_1 = boost::math::erfc_inv(x_1);
+  const double a_2 = boost::math::erfc_inv(x_2) - a_1;
+
+  // 2.82842712475 = 2*sqrt(2)
+  return (2.82842712475 * equivalent_surface_roughness * a_2) /
+         (M_PI * thermal_conductivity_gas * contact_radius_squared *
+          std::log(1 + a_2 / (a_1 + gas_parameter_m /
+                                      (2.82842712475 *
+                                       equivalent_surface_roughness))));
+}
+
+double
+calculate_interstitial_gas_macrogap_resistance(
+  const double harmonic_radius,
+  const double thermal_conductivity_gas,
+  const double contact_radius_squared,
+  const double gas_parameter_m)
+{
+  const double A =
+    2. * sqrt(harmonic_radius * harmonic_radius - contact_radius_squared);
+  const double S = 2. * harmonic_radius -
+                   contact_radius_squared / harmonic_radius + gas_parameter_m;
+
+  return 2.0 / (M_PI * thermal_conductivity_gas * (S * log(S / (S - A)) - A));
+}
+
+template <ContactType contact_type>
+void
+calculate_contact_thermal_conductance(
+  const double                  radius_one,
+  [[maybe_unused]] const double radius_two,
+  const double                  effective_youngs_modulus,
+  const double                  effective_real_youngs_modulus,
+  const double                  equivalent_surface_roughness,
+  const double                  equivalent_surface_slope,
+  const double                  effective_microhardness,
+  const double                  thermal_conductivity_one,
+  const double                  thermal_conductivity_two,
+  const double                  thermal_conductivity_gas,
+  const double                  gas_parameter_m,
+  const double                  normal_overlap,
+  const double                  normal_force_norm,
+  double                       &thermal_conductance)
+{
+  const double harmonic_conductivity =
+    harmonic_mean(thermal_conductivity_one, thermal_conductivity_two);
+  // For particle-wall contacts, it is as if the radius of the wall is infinite.
+  const double harmonic_radius = [&]() {
+    if constexpr (contact_type == ContactType::particle_floating_mesh)
+      return 2. * radius_one;
+    else
+      return harmonic_mean(radius_one, radius_two);
+  }();
```

Comment:
I'm not sure you need a lambda for this.
Most likely if you write:
const double harmonic_radius =  (contact_type==ContactType::particle_floating_mesh) ? 2*radius_one : harmonic_mean(radius_one,radius_two)  

This would be equally good and the compiler would figure it out. However, both are fine I believe :)